### PR TITLE
Feat/#337 data diff generation

### PR DESCRIPTION
**Pull request by ciaranschutte**
vast majority of changes are converting components to typescript/emotion and data schemas
main files for review:
https://github.com/icgc-argo/argo-docs/pull/340/files#diff-bbee10ede2cad8bee3608ecac1a795dc
https://github.com/icgc-argo/argo-docs/pull/340/files#diff-be7ea9d01b2e27d95025849d1a6115a8

### CODE REVIEW

File: scripts/manage-versions.ts [hunk 1]
```diff
@@ -26,10 +28,12 @@ const fs = require('fs');
 const argv = require('yargs').argv;
 const fse = require('fs-extra');
 const generateTreeData = require('./generateData');
+const generateDiffChanges = require('./generateDiffData').default;
```
Comment (hlminh2000) [inline scripts/manage-versions.ts:31]: all these requires can be converted to imports right?

File: scripts/generateDiffData/index.ts [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+  };
+};
+
+type Diff = {
+  [key: string]: {
+    [key in ChangeTypeName]: {
+      [key: string]: Field | DiffField;
+    };
+  };
+};
```
Comment (hlminh2000) [inline scripts/generateDiffData/index.ts:21]: Unused type?

File: scripts/generateDiffData/index.ts [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+    };
+    restrictionChanges?: {
+      regex?: Change<string>;
+      codelist?: Change<Array<string | number>>;
+      required?: Change<boolean>;
+      script?: Change<Array<string> | string>;
+      range?: Change<any>;
+    };
+  };
+};
```
Comment (hlminh2000) [inline scripts/generateDiffData/index.ts:176]: Unused type?

File: scripts/generateDiffData/index.test.ts [hunk 4]
```diff
@@ -0,0 +1,425 @@
+/*
```
Comment (hlminh2000) [inline scripts/generateDiffData/index.test.ts:1]: 👍  Thank you for the test! :) Should we add this to the jenkinsfile for test during CI?

File: scripts/manage-versions.ts [hunk 5]
```diff
@@ -61,8 +65,8 @@ function saveFiles(version, data) {
   const dataFile = `${schemaPath}/${version}.json`;
   const treeFile = `${schemaPath}/${version}_tree.json`;
   fse.writeJSONSync(dataFile, data);
-  const treeData = generateTreeData(data);
-  fse.writeJSONSync(treeFile, treeData);
+  // const treeData = generateTreeData(data);
+  // fse.writeJSONSync(treeFile, treeData);
```
Comment (hlminh2000) [inline scripts/manage-versions.ts:69]: worth keeping around or should we just remove?

File: scripts/manage-versions.ts [hunk 6]
```diff
@@ -77,8 +81,8 @@ function saveDataFiles(dictionary, versions) {
     currentVersion: versions[0],
   };
   fs.writeFileSync(dataFilename, JSON.stringify(content));
-  const treeData = generateTreeData(content.dictionary);
-  fse.writeJSONSync(dataFileTreeName, treeData);
+  //const treeData = generateTreeData(content.dictionary);
+  //fse.writeJSONSync(dataFileTreeName, treeData);
```
Comment (hlminh2000) [inline scripts/manage-versions.ts:85]: worth keeping around or should we just remove?

File: scripts/manage-versions.ts (hunk 5 above)
Comment (ciaranschutte) [reply scripts/manage-versions.ts:69]: worth keeping for now, i think tree will be worked on pretty soon

File: scripts/manage-versions.ts (hunk 6 above)
Comment (ciaranschutte) [reply scripts/manage-versions.ts:85]: worth keeping for now, i think tree will be worked on pretty soon

File: scripts/manage-versions.ts (hunk 1 above)
Comment (ciaranschutte) [reply scripts/manage-versions.ts]: yup! nice catch https://github.com/icgc-argo/argo-docs/pull/340/commits/b888f4d118c1811063fe9b321b7932a336b0fb71

**Comment by ciaranschutte** [comment]
> If we can actually use the types in the script that'd be really awesome, but if we can't fit it into available time that's ok for now.
> 

cleaned up types a little but pushed for time. will leave this for now.

File: scripts/generateDiffData/index.test.ts (hunk 4 above)
Comment (ciaranschutte) [reply scripts/generateDiffData/index.test.ts:1]: @hlminh2000 yes, great idea. Will do in different PR due to current Jenkins issues.

**Pull request merged**